diesel_derives2/src/field.rs

```diff
@@ -35,7 +36,15 @@ impl Field {
         self.column_name_from_attribute
             .unwrap_or_else(|| match self.name {
                 FieldName::Named(x) => x,
-                _ => panic!("All fields of tuple structs must be annotated with `#[column_name]`"),
+                _ => {
+                    self.ty
+                        .span()
+                        .error(
+                            "All fields of tuple structs must be annotated with `#[column_name]`",
+                        )
+                        .emit();
+                    "unknown_column".into()
```

Comment:
Maybe this should explicitly be replaced by a call to `unreachable!()`

Reply:
It's not unreachable.